tests/unit/test_diffsync_model_flags.py

```diff
@@ -0,0 +1,196 @@
+"""Unit tests for the DiffSyncModel flags.
+
+Copyright (c) 2020-2021 Network To Code, LLC <info@networktocode.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import pytest
+
+from diffsync.enum import DiffSyncModelFlags
+from diffsync.exceptions import ObjectNotFound
+
+
+def test_diffsync_diff_with_skip_unmatched_src_flag_on_models(backend_a, backend_a_with_extra_models):
+    # Validate that there are 2 extras objects out of the box
+    diff = backend_a.diff_from(backend_a_with_extra_models)
+    assert diff.summary() == {"create": 2, "update": 0, "delete": 0, "no-change": 23}
+
+    # Check that only 1 object is affected by the flag
+    backend_a_with_extra_models.get(
+        backend_a_with_extra_models.site, "lax"
+    ).model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_SRC
+    diff = backend_a.diff_from(backend_a_with_extra_models)
+    assert diff.summary() == {"create": 1, "update": 0, "delete": 0, "no-change": 23}
```

Comment:
Seems weird to me that setting the flag makes this model "disappear" from the summary counts. Offhand I would expect that the diff summary should still show a "create" for that model - but if that's not the case, should it instead show as a "no-change" or something?

Reply:
This behavior is inline with the behavior of the global flags `SKIP_UNMATCHED_*`. 
I've updated the unit tests for the global flags to confirm that.
I kind of agree with you but for now I think it's better to be consistent rather than introducing what could be considered a breaking change.

Reply:
Would it be possible and appropriate to add a "skipped" key to the diff summary? I'm okay with opening that as a separate enhancement request if it's non-trivial to implement.

Reply:
Good idea, I created it https://github.com/networktocode/diffsync/issues/89